### PR TITLE
Improve extracting of the bundle

### PR DESCRIPTION
**Pull request by saberduck**
`ZipInputStream` should be closed after extracting and we should verify that we are extracting within target directory

### CODE REVIEW

File: sonar-css-plugin/src/main/java/org/sonar/css/plugin/server/bundle/Zip.java [hunk 1]
```diff
@@ -19,34 +19,44 @@
...
-      zip.closeEntry();
-      entry = zip.getNextEntry();
     }
   }
+
+  private static Path entryPath(Path targetPath, ZipEntry entry) {
+    Path entryPath = targetPath.resolve(entry.getName()).normalize();
+    if (!entryPath.startsWith(targetPath)) {
+      throw new IllegalStateException("Archive entry " + entry.getName() + " is not within " + targetPath);
+    }
```
Comment (yassin-kammoun-sonarsource) [inline sonar-css-plugin/src/main/java/org/sonar/css/plugin/server/bundle/Zip.java:59]: Can we have a test case for this ?

Comment (saberduck) [reply sonar-css-plugin/src/main/java/org/sonar/css/plugin/server/bundle/Zip.java:59]: challenge accepted!

Comment (yassin-kammoun-sonarsource) [reply sonar-css-plugin/src/main/java/org/sonar/css/plugin/server/bundle/Zip.java:59]: Great!

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
Looks great to me.

**Pull request merged**